### PR TITLE
Added test for SolutionTransfer with hp-refinement and manually set flags.

### PR DESCRIPTION
**Pull request by marcfehling**
We did not have a simple test yet that checks if SolutionTransfer works when changing the FE indices.

@peterrum -- FYI

### CODE REVIEW

**Review by peterrum — APPROVED** [review]
Thanks!

File: tests/mpi/solution_transfer_11.cc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  tria.execute_coarsening_and_refinement();
+
+  dofh.distribute_dofs(fes);
+  locally_owned_dofs    = dofh.locally_owned_dofs();
+  locally_relevant_dofs = DoFTools::extract_locally_relevant_dofs(dofh);
+
+  solution.reinit(locally_owned_dofs,
+                  locally_relevant_dofs,
+                  dofh.get_communicator());
+  soltrans.interpolate(solution);
```
Comment (bangerth) [inline tests/mpi/solution_transfer_11.cc:122]: This is the end of the test. Would it make sense to compute some diagnostic information from the so-interpolated vector? Say its l1 norm or some something one could output? Right now, all you test is that the test doesn't abort.

**Review by bangerth — CHANGES_REQUESTED** [review]

File: tests/mpi/solution_transfer_11.cc (hunk 1 above)
Comment (marcfehling) [reply tests/mpi/solution_transfer_11.cc:126]: I've added the output of the l1 norm pre and post refinement.

Comment (bangerth) [reply tests/mpi/solution_transfer_11.cc:126]: Great, thanks!

**Review by bangerth — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Pull request merged**